### PR TITLE
alt

### PR DESCRIPTION
**Pull request by wxtim**

### CODE REVIEW

File: cylc/rose/utilities.py [hunk 1]
```diff
@@ -192,13 +192,38 @@ def identify_templating_section(config_node):
             "You should not define more than one templating section. "
             f"You defined:\n\t{'; '.join(defined_sections)}"
         )
-    elif 'jinja2:suite.rc' in defined_sections:
-        templating = 'jinja2:suite.rc'
-    elif 'empy:suite.rc' in defined_sections:
-        templating = 'empy:suite.rc'
     else:
+        return _identify_templating_section(defined_sections)
```
Comment (oliver-sanders) [inline cylc/rose/utilities.py]:
```suggestion
        return _identify_templating_section(list(defined_sections)[0:1])
```

File: cylc/rose/utilities.py [hunk 2]
```diff
@@ -192,13 +192,38 @@ def identify_templating_section(config_node):
...
+            * A section name
+            * None
+    """
+    templating = None
+    if sections:
+        if isinstance(sections, set):
+            section = sections.pop() if sections else []
+        else:
+            section = sections
+
```
Comment (oliver-sanders) [inline cylc/rose/utilities.py]:
```suggestion
def _identify_templating_section(
    sections: Optional[str, None] = None,
    with_brackets: Optional[bool] = False
) -> str:
    """Return a full template section string."""
    templating = None
```

File: cylc/rose/utilities.py [hunk 3]
```diff
@@ -192,13 +192,38 @@ def identify_templating_section(config_node):
...
         )
-    elif 'jinja2:suite.rc' in defined_sections:
-        templating = 'jinja2:suite.rc'
-    elif 'empy:suite.rc' in defined_sections:
-        templating = 'empy:suite.rc'
     else:
+        return _identify_templating_section(defined_sections)
+
+
+def _identify_templating_section(
```
Comment (oliver-sanders) [inline cylc/rose/utilities.py]: This shouldn't have an underscore prefix.

Underscore prefix means, don't use this outside of the scope it's defined in e.g:

* Underscore-prefixed method: only the class should use this
* Underscore-prefixed function: only the module should use this
* Underscore-prefixed module: only the package should use this

**Review by oliver-sanders — APPROVED** [review]

File: cylc/rose/utilities.py [hunk 4]
```diff
@@ -192,13 +192,25 @@ def identify_templating_section(config_node):
...
         )
-    elif 'jinja2:suite.rc' in defined_sections:
-        templating = 'jinja2:suite.rc'
-    elif 'empy:suite.rc' in defined_sections:
-        templating = 'empy:suite.rc'
     else:
+        return _identify_templating_section(list(defined_sections)[0:1])
+
+
+def _identify_templating_section(
```
Comment (wxtim) [inline cylc/rose/utilities.py]:
```suggestion
        return id_templating_section(list(defined_sections)[0:1])


def id_templating_section(
```

File: tests/unit/test_config_node.py [hunk 5]
```diff
@@ -252,6 +253,20 @@ def test_identify_templating_section(node_, expect, raises):
...
+    'input_, expect',
+    (
+        ([None], 'template variables'),
+        (['jinja2'], 'jinja2:suite.rc'),
+        ([None, True], '[template variables]'),
+        (['jinja2', True], '[jinja2:suite.rc]'),
+    )
+)
+def test__identify_templating_section(input_, expect):
+    assert _identify_templating_section(*input_) == expect
```
Comment (wxtim) [inline tests/unit/test_config_node.py]:
```suggestion
def test_identify_templating_section(input_, expect):
    assert id_templating_section(*input_) == expect
```

**Pull request merged**